src/cursor.rs

```diff
@@ -19,29 +18,29 @@ impl CursorPosition {
     pub fn move_left(&mut self) {
         self.x = cmp::max(1, self.x - 1);
     }
-    pub fn move_right(&mut self, lines: &Vec<String>) {
+    pub fn move_right(&mut self, lines: &[String]) {
```

Comment:
Interesting. Is this still underlying a vector or an array?

Reply:
This was to fix a warning by the linter (clippy). I did not have to change the caller of the function so the two notation are equivalent: a borrowed collection of String.
Modify that back and run clippy you will see the warning.